airflow-core/src/airflow/ui/src/i18n/check_translations_completeness.py

```diff
@@ -0,0 +1,333 @@
+            sys.exit(2)
+        else:
+            filtered_summary = {}
+            for filename, diff in summary.items():
+                filtered_summary[filename] = LocaleSummary(
+                    missing_keys={language: diff.missing_keys.get(language, [])},
+                    extra_keys={language: diff.extra_keys.get(language, [])},
+                )
+            lang_diff = print_language_summary(
+                [lf for lf in locale_files if lf.locale == language], filtered_summary, console
+            )
+            found_difference = found_difference or lang_diff
+            if add_missing:
+                add_missing_translations(language, filtered_summary, console)
+    else:
+        lang_diff = print_language_summary(locale_files, summary, console)
+        found_difference = found_difference or lang_diff
+    if not found_difference:
+        console.print("[green]All translations are complete and consistent![/green]")
+    if found_difference:
+        sys.exit(1)
+
+
+def add_missing_translations(language: str, summary: dict[str, LocaleSummary], console: Console):
+    """
+    Add missing translations for the selected language.
+
+    It does it by copying them from English and prefixing with 'TODO: translate:'.
+    """
+    for filename, diff in summary.items():
+        missing_keys = diff.missing_keys.get(language, [])
+        if not missing_keys:
+            continue
+        en_path = LOCALES_DIR / "en" / filename
+        lang_path = LOCALES_DIR / language / filename
+        try:
+            en_data = load_json(en_path)
+        except Exception as e:
+            console.print(f"[red]Failed to load English file {en_path}: {e}[/red]")
+            continue
+        try:
+            lang_data = load_json(lang_path)
+        except Exception as e:
+            console.print(f"[red]Failed to load {language} file {language}: {e}[/red]")
+            sys.exit(1)
+
+        # Helper to recursively add missing keys
+        def add_keys(src, dst, prefix=""):
+            for k, v in src.items():
+                full_key = f"{prefix}.{k}" if prefix else k
+                if full_key in missing_keys:
+                    if isinstance(v, dict):
+                        dst[k] = {}
+                        add_keys(v, dst[k], full_key)
+                    else:
+                        dst[k] = f"TODO: translate: {v}"
+                else:
+                    if isinstance(v, dict):
+                        if k not in dst or not isinstance(dst[k], dict):
+                            dst[k] = {}
+                        add_keys(v, dst[k], full_key)
+
+        add_keys(en_data, lang_data)
+        # Write back to file, preserving order
+        lang_path.parent.mkdir(parents=True, exist_ok=True)
+        with open(lang_path, "w", encoding="utf-8") as f:
+            json.dump(lang_data, f, ensure_ascii=False, indent=2, sort_keys=True)
```

Comment:
Ah and I just realized that newly written files are missing a newline, so we need to add this not to fail on pre-commit afterwards;
```suggestion
            json.dump(lang_data, f, ensure_ascii=False, indent=2, sort_keys=True)
            f.write("\n")  # Ensure newline at the end of the file
```

Reply:
Done.